src/foundry/foundry.js/pixi/polygons/pointSourcePolygon.d.ts

```diff
@@ -0,0 +1,97 @@
+interface PointSourcePolygonConfig {
+  /** The type of polygon being computed */
+  type: foundry.CONST.WALL_RESTRICTION_TYPES;
+
+  /**
+   * The angle of emission, if limited
+   * @defaultValue `360`
+   */
+  angle: number;
+
+  /** The desired density of padding rays, a number per PI */
+  density: number;
+
+  /** A limited radius of the resulting polygon */
+  radius: number;
+
+  /** The direction of facing, required if the angle is limited */
+  rotation: number;
+
+  /** Display debugging visualization and logging for the polygon */
+  debug: boolean;
+
+  /** Is this polygon constrained by any walls? */
+  walls: boolean;
+
+  /** The object (if any) that spawned this polygon. */
+  source: PointSource;
+}
+
+/**
+ * An extension of the default PIXI.Polygon which is used to represent the line of sight for a point source.
+ */
+declare abstract class PointSourcePolygon extends PIXI.Polygon {
+  /**
+   * The origin point of the source polygon.
+   */
+  origin: Point;
+
+  /**
+   * The configuration of this polygon.
+   */
+  config: PointSourcePolygonConfig;
+
+  /**
+   * A cached array of SightRay objects used to compute the polygon.
+   */
+  rays: SightRay[];
+
+  /**
+   * Compute the rectangular bounds for the Polygon.
+   * @param points - The initially provided array of coordinates
+   * @returns The computed Rectangular bounds
+   */
+  protected _getBounds(points: number[]): PIXI.Rectangle;
+
+  /**
+   * Benchmark the performance of polygon computation for this source
+   * @param iterations - The number of test iterations to perform
+   * @param args       - Arguments passed to the compute method
+   */
+  static benchmark(
+    iterations: number,
+    ...args: Parameters<typeof PointSourcePolygon['create']>
+  ): ReturnType<typeof foundry.utils.benchmark>;
+
+  /**
+   * Compute the polygon given a point origin and radius
+   * @param origin - The origin source point
+   * @param config - Configuration options which customize the polygon computation
+   *                 (default: `{}`)
+   * @returns The computed polygon instance
+   */
+  static create(origin: Point, config?: Partial<PointSourcePolygonConfig>): ReturnType<PointSourcePolygon['compute']>;
```

Comment:
I don't think the `type` property of `config` may be optional, so `config` also cannot be optional, I guess? It's documented in foundry as being optional, defaulting to `{}` but it's not implemented like that, as far as I can see (potentially worth a bug report).